tests/snos.rs

```diff
@@ -98,3 +118,35 @@ fn test_different_outputs() {
     .expect("Couldn't run program");
     assert_python_and_rust_output_match("build/different_output.json", virtual_machine);
 }
+
+#[test]
+fn load_compiled_classes_facts_test() {
```

Comment:
could we externalize the logic of this function by extending the SnOsRunner ? I feel like we're repeating ourselves here and makes it hard to debug